.circleci/config.yml

```diff
@@ -1,71 +1,131 @@
-version: 2
+version: 2.1
 
-jobs:
-  build:
+
+executors:
+  node-latest:
+    working_directory: ~/repo
     docker:
       - image: circleci/node:latest
+
+jobs:
+  init:
+    executor: node-latest
     steps:
       - checkout
       - run:
-          name: Update npm
-          command: 'sudo npm install -g npm@latest'
-      - restore_cache:
-          key: git-sha-{{ .Revision }}
+          name: install ci tools
+          command: 'mkdir -p ~/.build_tools/node && npm install --prefix ~/.build_tools/node kl-ci-tools'
       - run:
-          name: NPM Install
-          command: npm install
+          name: update dependencies if releasing
+          command: |
+            if [[ "$CIRCLE_BRANCH" =~ release/v.* ]]; then
+              ~/.build_tools/node/node_modules/kl-ci-tools/dist/bin.js deps | xargs -I[] npm install []@latest
+            fi
       - run:
-          name: NPM Build
+          name: npm install
+          command: npm i
+      - run:
+          name: npm build
           command: npm run build
       - save_cache:
-          key: git-sha-{{ .Revision }}
+          key: git-sha-{{ .Revision }}{{ .Branch }}
           paths:
-            - node_modules
             - dist
+            - package.json
+            - node_modules
+            - ~/.build_tools
+
   test:
-    docker:
-      - image: circleci/node:latest
+    executor: node-latest
     steps:
       - checkout
       - restore_cache:
-          key: git-sha-{{ .Revision }}
+          key: git-sha-{{ .Revision }}{{ .Branch }}
       - run:
-          name: Lint
+          name: lint
           command: npm run lint
       - run:
-          name: Test
+          name: test
           command: npm test
+
   publish-dev:
-    docker:
-      - image: circleci/node:latest
+    executor: node-latest
     steps:
       - checkout
       - restore_cache:
-          key: git-sha-{{ .Revision }}
+          key: git-sha-{{ .Revision }}{{ .Branch }}
       - run:
-          name: Set Develop Version
-          command: git rev-parse develop | cut -c1-10 | xargs -I[] ./node_modules/package-tagger/index.js dev.[]
+          name: set develop version
+          command: git rev-parse develop | cut -c1-10 | xargs -I[] ~/.build_tools/node/node_modules/kl-ci-tools/dist/bin.js tag dev.[]
       - run:
-          name: Authenticate with registry
+          name: authenticate with registry
           command: echo "//registry.npmjs.org/:_authToken=$npm_TOKEN" > .npmrc
       - run:
-          name: Publish package
+          name: publish package
           command: npm publish --tag develop
 
+  publish-latest:
+    executor: node-latest
+    steps:
+      - checkout
+      - restore_cache:
+          key: git-sha-{{ .Revision }}{{ .Branch }}
+      - run:
+          name: setup git config
+          command: git config user.email "dev@mazzaroth.io" && git config user.name "CricleCI" && git config --global core.editor "\usr\local\bin\subl -n -w"
```

Comment:
Misspelled "CircleCI"

Reply:
Hahaha, I think this one is probably copy and pasted everywhere at this point. I'll fix it thanks.

Reply:
Lol yeah I just now noticed it